extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -459,6 +469,67 @@ void validateConfigPropertiesInjectionPoints(
         toRegister.forEach(configProperties::produce);
     }
 
+    @BuildStep
+    void warnStaticInitInjectionPoints(
+            CombinedIndexBuildItem indexBuildItem,
+            ValidationPhaseBuildItem validationPhase,
+            List<ConfigClassBuildItem> configClasses,
+            BuildProducer<RunTimeConfigurationDefaultBuildItem> runTimeConfigurationDefault) {
+
+        // Add here annotated classes that are initialized during static init
+        Set<DotName> declaringClassCandidates = new HashSet<>();
+        declaringClassCandidates.add(DotName.createSimple("javax.ws.rs.ext.Provider"));
```

Comment:
For RESTEasy Reactive, I don't think think is a problem actually...

Reply:
In RESTEasy Reactive, Providers are not initialized at static init? Ok, let me check and add a couple of tests there.

Reply:
In RESTEasy Reactive they are just normal beans

Reply:
Confirmed. Interesting the different behavior between both. I'll add a build item.